src/code/z_map_disp.c

```diff
@@ -1,109 +1,1881 @@
+            gDPLoadTextureBlock_4b(OVERLAY_DISP++, texture, G_IM_FMT_IA, texWidth, texHeight, 0,
+                                   G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                                   G_TX_NOLOD, G_TX_NOLOD);
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80105B34.s")
+        case MAPDATA_DRAW_3:
+            gDPSetCombineLERP(OVERLAY_DISP++, 0, 0, 0, PRIMITIVE, 0, 0, 0, TEXEL0, 0, 0, 0, PRIMITIVE, 0, 0, 0, TEXEL0);
+            gDPLoadTextureBlock_4b(OVERLAY_DISP++, texture, G_IM_FMT_I, texWidth, texHeight, 0,
+                                   G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                                   G_TX_NOLOD, G_TX_NOLOD);
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80105C40.s")
+        default:
+        case MAPDATA_DRAW_0:
+            gDPSetCombineMode(OVERLAY_DISP++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+            gDPLoadTextureBlock_4b(OVERLAY_DISP++, texture, G_IM_FMT_I, texWidth, texHeight, 0,
+                                   G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                                   G_TX_NOLOD, G_TX_NOLOD);
+            break;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80105FE0.s")
+    s = (mapDataRoom->flags & 1) ? (texWidth - 1) << 5 : 0;
+    t = (mapDataRoom->flags & 2) ? 0 : (texHeight - 1) << 5;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80106408.s")
+    dsdx_temp = ((mapDataRoom->flags & 1) ? -1 : 1) * (1 << 10);
+    dtdy_temp = ((mapDataRoom->flags & 2) ? 1 : -1) * (1 << 10);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80106450.s")
+    dsdx = (mapDataRoom->flags & 1) ? dsdx_temp & 0xFFFF : dsdx_temp;
+    dtdy = (mapDataRoom->flags & 2) ? dtdy_temp : dtdy_temp & 0xFFFF;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_801064CC.s")
+    gSPTextureRectangle(OVERLAY_DISP++, x << 2, y << 2, (texWidth + x) << 2, (y + texHeight) << 2, G_TX_RENDERTILE, s,
+                        t, dsdx, dtdy);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80106530.s")
+    CLOSE_DISPS(play->state.gfxCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_8010657C.s")
+// Tests if the map data should be rotated 180 degrees
+// SCENE_35TAKI is the only scene with data flipped in this manner.
+s32 MapDisp_IsDataRotated(PlayState* play) {
+    if (play->sceneId == SCENE_35TAKI) {
+        return true;
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80106644.s")
+s32 MapDisp_CanDrawDoors(PlayState* play) {
+    if ((gSaveContext.save.entrance == ENTRANCE(ROMANI_RANCH, 0)) && (Cutscene_GetSceneLayer(play) != 0)) {
+        return false;
+    }
+    return true;
+}
```

Comment:
It's to hide them when doing the [archery training stuff](https://www.youtube.com/watch?v=5cgcED2pisw), but it doesn't seem to apply to aliens because they spawn in on a playable scene setup.

Reply:
Oh, this is checking if you can draw doors on the minimap as white dots. Makes sense